code/game/mecha/equipment/tools/janitor_tools.dm

```diff
@@ -0,0 +1,363 @@
+		var/obj/structure/reagent_dispensers/watertank/WT = target
+		WT.reagents.trans_to(src, 1000)
+		occupant_message("<span class='notice'>Mop refilled.</span>")
+		playsound(chassis, 'sound/effects/refill.ogg', 50, TRUE, -6)
+	else
+		if(get_dist(chassis, target) > 2)
+			return
+		if(reagents.total_volume > 0)
+			if(COOLDOWN_FINISHED(src, mop_sound_cooldown))
+				playsound(loc, pick('sound/weapons/mopping1.ogg', 'sound/weapons/mopping2.ogg'), 30, TRUE, -1)
+				COOLDOWN_START(src, mop_sound_cooldown, MOP_SOUND_CD)
+			// 3x3 mopping area
+			var/turf/target_turf = get_turf(target)
+			if(!istype(target_turf) || iswallturf(target_turf))
+				return
+			chassis.occupant.visible_message("<span class='warning'>[chassis] begins to mop \the [target_turf] with \the [src].</span>", "<span class='warning'>You begin to mop \the [target_turf] with \the [src].</span>")
+			if(do_after(chassis.occupant, mop_speed, target = target))
+				for(var/turf/current_target_turf in view(1, target))
+					current_target_turf.cleaning_act(chassis.occupant, src, mop_speed, "mop", ".", skip_do_after = TRUE)
+				to_chat(chassis.occupant, "<span class='notice'>You mop \the [target].</span>")
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/post_clean(atom/target, mob/user)
+	var/turf/T = get_turf(target)
+	if(issimulatedturf(T))
+		reagents.reaction(T, REAGENT_TOUCH, 10)	// 10 is the multiplier for the reaction effect. 10 is needed to properly wet a floor.
+	reagents.remove_any(1)	// reaction() doesn't use up the reagents
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/can_clean()
+	if(reagents.has_reagent("water", 1) || reagents.has_reagent("cleaner", 1) || reagents.has_reagent("holywater", 1))
+		return TRUE
+	else
+		return FALSE
+
+// Auto-regeneration of water. Takes energy.
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/process()
+	if(reagents.total_volume < 1000)
+		reagents.add_reagent(refill_reagent, refill_rate)
+		chassis.use_power(refill_cost)
+		update_equip_info()
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/get_equip_info()
+	var/output = ..()
+	if(output)
+		return "[output] \[<a href='byond://?src=[UID()];toggle_mode=1'>Refill [refill_enabled? "Enabled" : "Disabled"]</a>\] \[[reagents.total_volume]\]"
+
+/obj/item/mecha_parts/mecha_equipment/janitor/mega_mop/Topic(href,href_list)
+	..()
+	var/datum/topic_input/afilter = new (href, href_list)
+	if(afilter.get("toggle_mode"))
+		refill_enabled = !refill_enabled
+		if(refill_enabled)
+			START_PROCESSING(SSobj, src)
+		else
+			STOP_PROCESSING(SSobj, src)
+		update_equip_info()
+
+#undef MOP_SOUND_CD
+
+// Light Replacer
+/obj/item/mecha_parts/mecha_equipment/janitor/light_replacer
+	name = "\improper NT-12 illuminator"
+	desc = "A modified light replacer fit for an exosuit that zaps lights into place."
+	icon_state = "mecha_light_replacer"
+	equip_cooldown = 1.5 SECONDS
+	energy_drain = 100
+	range = MECHA_MELEE | 15
```

Comment:
What's going on here?

Reply:
Line 114 denotes the range of the module. In this case, it can be used in "melee" range, or with a range of 15 tiles.